@exceptions.str.1 = private unnamed_addr constant  [59 x i8] c"Exception in thread main kotlin.KotlinNullPointerException\00", align 1
declare void @llvm.memcpy.p0i8.p0i8.i64(i8* nocapture, i8* nocapture readonly, i64, i32, i1)
declare i8* @malloc_heap(i32)
declare i32 @printf(i8*, ...)
declare void @abort()
%class.Nothing = type { }
define weak i32 @while_test_2_Int(i32  %while_test_2.x) #0
{
%while_test_2.x.addr = alloca i32, align 4
store i32 %while_test_2.x, i32* %while_test_2.x.addr, align 4
%managed.unique.0.while_test_2.a = alloca i32, align 4
%var2 = load i32* %managed.unique.0.while_test_2.a, align 4
%var3 = load i32* %while_test_2.x.addr, align 4
store i32 %var3, i32* %managed.unique.0.while_test_2.a, align 4
br label %label.while.1
label.while.1:
%var4 = icmp slt i32 2, 3
br i1 %var4, label %label.while.2, label %label.while.3
label.while.2:
%var5 = load i32* %while_test_2.x.addr, align 4
%var6 = add nsw i32 %var5, 1
%var7 = load i32* %managed.unique.0.while_test_2.a, align 4
store i32 %var6, i32* %managed.unique.0.while_test_2.a, align 4
%var8 = load i32* %managed.unique.0.while_test_2.a, align 4
ret i32 %var8
br label %label.while.1
label.while.3:
unreachable
}

